internal/controllers/clusterextension_controller.go

```diff
@@ -550,21 +552,54 @@ func (r *ClusterExtensionReconciler) generateBundleDeploymentForUnpack(bundlePat
 				Image: &rukpakv1alpha2.ImageSource{
 					Ref:                   bundlePath,
 					InsecureSkipTLSVerify: isInsecureSkipTLSVerifySet(ce),
+					CertificateData:       certData,
 				},
 			},
 		},
 	}
 }
 
 func isInsecureSkipTLSVerifySet(ce *ocv1alpha1.ClusterExtension) bool {
-	if ce == nil {
+	if ce == nil || ce.Spec.RegistryTLS == nil {
 		return false
 	}
-	value, ok := ce.Annotations[bundleConnectionAnnotation]
-	if !ok {
-		return false
+	return ce.Spec.RegistryTLS.InsecureSkipTLSVerify
+}
+
+func (r *ClusterExtensionReconciler) getCertificateData(ctx context.Context, ce *ocv1alpha1.ClusterExtension) (string, error) {
+	if ce == nil || ce.Spec.RegistryTLS == nil || ce.Spec.RegistryTLS.CertificateSecretRef == nil {
+		return "", nil
```

Comment:
No error and no cert here?

Reply:
The certificate is optional, if it's not present, we don't use it.